src/WebJobs.Script/ManagedDependencies/ManagedDependencyProvider.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script;
+using Microsoft.Azure.WebJobs.Script.Grpc.Messages;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.ManagedDependencies
+{
+    internal class ManagedDependencyProvider : IManagedDependencyProvider
+    {
+        private const string ManagedDependenciesPropertiesName = "managedDependencies";
+
+        /// <summary>
+        /// Returns the managed dependecies meta data
+        /// </summary>
+        /// <param name="hostFilePath">The configuration path for the managed dependencies</param>
+        /// <param name="environment">The environment object reprenested as inteface <see cref="IEnvironment"/> </param>
+        /// <param name="managedDependencyOptions">Configurations options for managed dependencies</param>
+        /// <returns>The IEnumerable list of <see cref="Dependency"/></returns>
+        public IEnumerable<Dependency> GetDependencies(string hostFilePath,
+            IEnvironment environment,
+            IOptions<ManagedDependencyOptions> managedDependencyOptions)
+        {
+            if (string.IsNullOrWhiteSpace(hostFilePath)
```

Comment:
For a valid function app, host.json exists. Do not have to check for its existence.